lib/elixir/lib/supervisor.ex

```diff
@@ -516,6 +518,13 @@ defmodule Supervisor do
   @typedoc "Option values used by the `start*` functions"
   @type option :: {:name, name}
 
+  @typedoc "The supervisor flags returned on init"
+  @type sup_flags() :: %{
```

Comment:
Is `sup_flags` the best name? `sup` is redundant... Maybe `init_flags` is better? :)

Reply:
You are right, that type is public, so there isn't much we can do.

Reply:
No, let’s go with sup_flags. It is not my favorite name but not enough to warrant a deprecation IMO. :)